deeppavlov/models/sklearn/sklearn_component.py

```diff
@@ -0,0 +1,315 @@
+        save_path: save path for model, e.g. full name ``model_path/model.pkl`` \
+            or prefix ``model_path/model`` (still model will be saved to ``model_path/model.pkl``)
+        load_path: load path for model, e.g. full name ``model_path/model.pkl`` \
+            or prefix ``model_path/model`` (still model will be loaded from ``model_path/model.pkl``)
+        infer_method: string name of class method to use for infering model, \
+            e.g. ``predict``, ``predict_proba``, ``predict_log_proba``, ``transform``
+    """
+    def __init__(self, model_class: str,
+                 save_path: Union[str, Path] = None,
+                 load_path: Union[str, Path] = None,
+                 infer_method: str = "predict", **kwargs) -> None:
+        """
+        Initialize component with given parameters
+        """
+
+        super().__init__(save_path=save_path, load_path=load_path, **kwargs)
+        self.model_class = model_class
+        self.model_params = kwargs
+        self.model = None
+        self.pipe_params = {}
+        for required in ["in", "out", "fit_on", "main", "name"]:
+            self.pipe_params[required] = self.model_params.pop(required, None)
+
+        self.load()
+        self.infer_method = infer_method
+
+    def fit(self, *args) -> None:
+        """
+        Fit model on the given data
+
+        Args:
+            *args: list of x-inputs and, optionally, one y-input (the last one) to fit on.
+                Possible input (x0, ..., xK, y) or (x0, ..., xK) '
+                where K is the number of input data elements (the length of list ``in`` from config). \
+                In case of several inputs (K > 1) input features will be stacked. \
+                For example, one has x0: (n_samples, n_features0), ..., xK: (n_samples, n_featuresK), \
+                then model will be trained on x: (n_samples, n_features0 + ... + n_featuresK).
+
+        Returns:
+            None
+        """
+        n_inputs = len(self.pipe_params["in"]) if isinstance(self.pipe_params["in"], list) else 1
+        x_features = self.compose_input_data(args[:n_inputs])
+        if len(args) > n_inputs:
+            y_ = np.squeeze(np.array(args[-1]))
+        else:
+            y_ = None
+
+        try:
+            log.info("Fitting model {}".format(self.model_class))
+            self.model.fit(x_features, y_)
+        except TypeError or ValueError:
+            try:
+                if issparse(x_features):
+                    log.info("Converting input for model {} to dense array".format(self.model_class))
+                    self.model.fit(x_features.todense(), y_)
+                else:
+                    log.info("Converting input for model {} to sparse array".format(self.model_class))
+                    self.model.fit(csr_matrix(x_features), y_)
+            except:
+                raise ConfigError("Can not fit on the given data".format(self.model_class))
```

Comment:
I think, it would be better to not `try, except` as the exception message will be lost and it will be harder to debug

Reply:
ok.fixed